mesonbuild/modules/__init__.py

```diff
@@ -87,15 +87,15 @@ def get_include_args(self, include_dirs: T.Iterable[T.Union[str, build.IncludeDi
     def find_program(self, prog: T.Union[mesonlib.FileOrString, T.List[mesonlib.FileOrString]],
                      required: bool = True,
                      version_func: T.Optional[T.Callable[[T.Union[ExternalProgram, build.Executable, OverrideProgram]], str]] = None,
-                     wanted: T.Optional[str] = None, silent: bool = False,
+                     wanted: str = '', silent: bool = False,
```

Comment:
Hmm, interesting question which one should be fixed :P since interpreter.py is typed up as `wanted: T.Union[str, T.List[str]] = '',` but checks via `if wanted:`. So do we want to pass an empty string around, or None? Both behave the same as far as find_program_impl() is concerned.

None is a true singleton...